src/Diagnostics.DataProviders/TokenService/SearchServiceTokenService.cs

```diff
@@ -0,0 +1,27 @@
+﻿using System;
+using Diagnostics.DataProviders.DataProviderConfigurations;
+using Microsoft.IdentityModel.Clients.ActiveDirectory;
+
+namespace Diagnostics.DataProviders.TokenService
+{
+    public class SearchServiceTokenService : TokenServiceBase
+    {
+        private static readonly Lazy<SearchServiceTokenService> instance = new Lazy<SearchServiceTokenService>(() => new SearchServiceTokenService());
```

Comment:
You'll want a private constructor to make this a singleton. Best to do a `sealed` class as well